addons/google_calendar/models/google_calendar.py

```diff
@@ -647,9 +647,16 @@ def create_new_events(self):
             ('event_id.final_date', '>', fields.Datetime.to_string(self.get_minTime())),
         ])
         for att in my_attendees:
-            other_google_ids = [other_att.google_internal_event_id for other_att in att.event_id.attendee_ids if
-                                other_att.google_internal_event_id and other_att.id != att.id and not other_att.google_internal_event_id.startswith('_')]
-            for other_google_id in other_google_ids:
+            # Get the attendees of the same event with google id set
+            other_attendees = att.event_id.attendee_ids.filtered(lambda other_att: other_att.google_internal_event_id and other_att.id != att.id and not other_att.google_internal_event_id.startswith('_'))
+            if att.partner_id in other_attendees.mapped('partner_id'):
+                # After a contacts merge there may be events
+                # with multiple attendee records of the same partner.
+                # Deleting duplicate attendee to avoid raising error on constraint google_id_uniq
+                att.unlink()
```

Comment:
Can a duplicate has some information that is not present in the attendee record we keep? If yes, a comment with TODO should be enough, because merging attendees seems too complicated for stable.

The rest looks ok.

Reply:
Attendee fields, with calendar and google_calendar installed are

* state, state of the participation, this might differ for whatever reason but it doesn't seem a great loss, by default it is 'pending'
* common_name, derived from the partner info (name/email), will hold old contact info, should be safe to delete if the contacts were merged on purpose to 'delete' the old one
* partner_id, this will be equal by definition
* email, same reasoning as common_name
* availability, Free/Busy, like state
* access_token, at least the destination contact of the merge is 'internal users' (otherwise they would not have access to google calendar sync) this can be safely lost
* event_id, equal by definition
* google_internal_event_id, in the deleted attendee is False, while there is another attendee holding the value
* oe_synchro_date, should be older in deleted attendee

So we are technically losing the attendee participation status, but related to the old contact, so this should be acceptable

Reply:
If it's only status, then it should be fine